webapp-django/crashstats/crashstats/static/crashstats/js/socorro/correlation.js

```diff
@@ -97,4 +97,169 @@ $(document).ready(function () {
+
+    function getDataURL(product) {
+        if (product === 'Firefox') {
+            return 'https://analysis-output.telemetry.mozilla.org/top-signatures-correlations/data/';
+        } else if (product === 'FennecAndroid') {
+            return 'https://analysis-output.telemetry.mozilla.org/top-fennec-signatures-correlations/data/';
+        } else {
+            throw new Error('Unknown product: ' + product);
+        }
+    }
+
+    function loadChannelsData(product) {
+        if (correlationData[product]) {
+            return Promise.resolve();
+        }
+
+        return fetch(getDataURL(product) + 'all.json.gz')
+        .then(response => response.json())
+        .then(totals => {
+            correlationData[product] = {
+                'date': totals['date'],
+            };
+
+            for (let ch of ['release', 'beta', 'aurora', 'nightly']) {
+                correlationData[product][ch] = {
+                    'total': totals[ch],
+                    'signatures': {},
+                }
+            }
+        });
+    }
+
+    function loadCorrelationData(signature, channel, product) {
+        return loadChannelsData(product)
+        .then(() => {
+            if (signature in correlationData[product][channel]['signatures']) {
+                return;
+            }
+
+            return sha1(signature)
+            .then(sha1signature => {
+              return sha1signature;
+            })
+            .then(sha1signature => fetch(getDataURL(product) + channel + '/' + sha1signature + '.json.gz'))
+            .then(response => response.json())
+            .then(data => {
+                correlationData[product][channel]['signatures'][signature] = data;
+            });
+        })
+        .catch(() => {})
+        .then(() => correlationData);
+    }
+
+    function itemToLabel(item) {
+        return Object.getOwnPropertyNames(item)
+        .map(key => key + ' = ' + item[key])
+        .join(' ∧ ');
+    }
+
+    function toPercentage(num) {
+        let result = (num * 100).toFixed(2);
+
+        if (result == '100.00') {
+          return '100.0';
+        }
+
+        if (result.substring(0, result.indexOf('.')).length == 1) {
+          return '0' + result;
+        }
+
+        return result;
+    }
+
+  function confidenceInterval(count1, total1, count2, total2) {
+    let prop1 = count1 / total1;
+    let prop2 = count2 / total2;
+    let diff = prop1 - prop2;
+
+    // Wald 95% confidence interval for the difference between the proportions.
+    let standard_error = Math.sqrt(prop1 * (1 - prop1) / total1 + prop2 * (1 - prop2) / total2);
+    let ci = [diff - 1.96 * standard_error, diff + 1.96 * standard_error];
+
+    // Yates continuity correction for the confidence interval.
+    let correction = 0.5 * (1.0 / total1 + 1.0 / total2);
+
+    return [ci[0] - correction, ci[1] + correction];
+  }
+
+  function sortCorrelationData(correlationData, total_reference, total_group) {
```

Comment:
I'll add some comments.

It sorts the results by number of attributes first (e.g. platform=Windows will come before platform=Windows && graphic_card=NVIDIA), because they are often the most interesting results.

Then, it sorts by percentage difference (e.g. if platform=Windows is 100% in the signature and 60% overall and graph_card=NVIDIA is 40% in the signature and 25% overall, platform=Windows will come first).